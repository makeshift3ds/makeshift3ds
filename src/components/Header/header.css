@import '../../assets/css/vars.css';

header {
  background-image: url('http://res.cloudinary.com/makeshift3ds/image/upload/header-background_scn7px.jpg');
  background-position: center bottom;
  border-bottom: 1px solid #2e3b3f;
  height: 385px;

  & #header-content-container {
    display: flex;
    height: 90%;

    @media (max-width: 630px) {
      display: block;
    }
  }

  & #header-avatar {
    display: flex;
    padding: 1rem;

    @media (max-width: 630px) {
      display: block;
      text-align: center;
      margin-top: 3rem;
    }

    & img {
      border-radius: 50%;
      border: 2px solid #2e3b3f;
      filter: sepia(0.5);
      box-shadow: var(--shadow-level1);
      align-self: flex-end;
    }
  }

  & .header-content {
    flex-grow: 1;
    text-align: right;
    padding: 1rem;
    font-size: 2rem;
    color: #dac7b9;
    align-self: flex-end;
    transform: translateY(-2rem);

    @media (max-width: 630px) {
      font-size: 1.5rem;
      text-align: center;
      margin-top: 3rem;
    }
  }
}
