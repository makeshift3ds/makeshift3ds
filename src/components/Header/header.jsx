import React from 'react';
import css from './header.css';
import Nav from '../Nav/nav.jsx';

const Header = () => (
  <header>
    <Nav />
    <div id="header-content-container">
      <div id="header-avatar">
        <img
          src="http://res.cloudinary.com/makeshift3ds/image/upload/makeshift-header-avatar_omdbug.jpg"
          sizes="(max-width: 630px) 150px, 250px"
          srcSet="
		http://res.cloudinary.com/makeshift3ds/image/upload/w_150/makeshift-header-avatar_omdbug.jpg 150w,
		http://res.cloudinary.com/makeshift3ds/image/upload/w_250/makeshift-header-avatar_omdbug.jpg 250w"
          alt="Circle image of middle aged man smiling on opaque palm tree gradient background."
        />
      </div>
      <div className="header-content">Full Stack Engineer / Designer</div>
    </div>
  </header>
);

export default Header;
