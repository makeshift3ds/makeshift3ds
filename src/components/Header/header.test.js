import React from 'react';
import Header from './header.jsx';
import { shallow, mount } from 'enzyme';

describe('<Header />', () => {
  it('should render 1 <Header /> component', () => {
    const component = shallow(<Header />);
    expect(component).toHaveLength(1);
  });

  it('should render 1 avatar', () => {
    const component = shallow(<Header />);
    expect(component.find('#header-avatar')).toHaveLength(1);
  });

  it('should render header-content', () => {
    const component = shallow(<Header />);
    expect(component.find('.header-content').text()).toContain(
      'Full Stack Engineer / Designer'
    );
  });

  it('should match snapshot', () => {
    const component = shallow(<Header />);
    expect(component).toMatchSnapshot();
  });
});
