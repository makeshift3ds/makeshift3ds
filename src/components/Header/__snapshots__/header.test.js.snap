// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Nav />,
        <div
          id="header-content-container"
        >
          <div
            id="header-avatar"
          >
            <img
              alt="Circle image of middle aged man smiling on opaque palm tree gradient background."
              sizes="(max-width: 630px) 150px, 250px"
              src="http://res.cloudinary.com/makeshift3ds/image/upload/makeshift-header-avatar_omdbug.jpg"
              srcSet=" http://res.cloudinary.com/makeshift3ds/image/upload/w_150/makeshift-header-avatar_omdbug.jpg 150w, http://res.cloudinary.com/makeshift3ds/image/upload/w_250/makeshift-header-avatar_omdbug.jpg 250w"
            />
          </div>
          <div
            className="header-content"
          >
            Full Stack Engineer / Designer
          </div>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              id="header-avatar"
            >
              <img
                alt="Circle image of middle aged man smiling on opaque palm tree gradient background."
                sizes="(max-width: 630px) 150px, 250px"
                src="http://res.cloudinary.com/makeshift3ds/image/upload/makeshift-header-avatar_omdbug.jpg"
                srcSet=" http://res.cloudinary.com/makeshift3ds/image/upload/w_150/makeshift-header-avatar_omdbug.jpg 150w, http://res.cloudinary.com/makeshift3ds/image/upload/w_250/makeshift-header-avatar_omdbug.jpg 250w"
              />
            </div>,
            <div
              className="header-content"
            >
              Full Stack Engineer / Designer
            </div>,
          ],
          "id": "header-content-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="Circle image of middle aged man smiling on opaque palm tree gradient background."
                sizes="(max-width: 630px) 150px, 250px"
                src="http://res.cloudinary.com/makeshift3ds/image/upload/makeshift-header-avatar_omdbug.jpg"
                srcSet=" http://res.cloudinary.com/makeshift3ds/image/upload/w_150/makeshift-header-avatar_omdbug.jpg 150w, http://res.cloudinary.com/makeshift3ds/image/upload/w_250/makeshift-header-avatar_omdbug.jpg 250w"
              />,
              "id": "header-avatar",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Circle image of middle aged man smiling on opaque palm tree gradient background.",
                "sizes": "(max-width: 630px) 150px, 250px",
                "src": "http://res.cloudinary.com/makeshift3ds/image/upload/makeshift-header-avatar_omdbug.jpg",
                "srcSet": " http://res.cloudinary.com/makeshift3ds/image/upload/w_150/makeshift-header-avatar_omdbug.jpg 150w, http://res.cloudinary.com/makeshift3ds/image/upload/w_250/makeshift-header-avatar_omdbug.jpg 250w",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Full Stack Engineer / Designer",
              "className": "header-content",
            },
            "ref": null,
            "rendered": "Full Stack Engineer / Designer",
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Nav />,
          <div
            id="header-content-container"
          >
            <div
              id="header-avatar"
            >
              <img
                alt="Circle image of middle aged man smiling on opaque palm tree gradient background."
                sizes="(max-width: 630px) 150px, 250px"
                src="http://res.cloudinary.com/makeshift3ds/image/upload/makeshift-header-avatar_omdbug.jpg"
                srcSet=" http://res.cloudinary.com/makeshift3ds/image/upload/w_150/makeshift-header-avatar_omdbug.jpg 150w, http://res.cloudinary.com/makeshift3ds/image/upload/w_250/makeshift-header-avatar_omdbug.jpg 250w"
              />
            </div>
            <div
              className="header-content"
            >
              Full Stack Engineer / Designer
            </div>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                id="header-avatar"
              >
                <img
                  alt="Circle image of middle aged man smiling on opaque palm tree gradient background."
                  sizes="(max-width: 630px) 150px, 250px"
                  src="http://res.cloudinary.com/makeshift3ds/image/upload/makeshift-header-avatar_omdbug.jpg"
                  srcSet=" http://res.cloudinary.com/makeshift3ds/image/upload/w_150/makeshift-header-avatar_omdbug.jpg 150w, http://res.cloudinary.com/makeshift3ds/image/upload/w_250/makeshift-header-avatar_omdbug.jpg 250w"
                />
              </div>,
              <div
                className="header-content"
              >
                Full Stack Engineer / Designer
              </div>,
            ],
            "id": "header-content-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="Circle image of middle aged man smiling on opaque palm tree gradient background."
                  sizes="(max-width: 630px) 150px, 250px"
                  src="http://res.cloudinary.com/makeshift3ds/image/upload/makeshift-header-avatar_omdbug.jpg"
                  srcSet=" http://res.cloudinary.com/makeshift3ds/image/upload/w_150/makeshift-header-avatar_omdbug.jpg 150w, http://res.cloudinary.com/makeshift3ds/image/upload/w_250/makeshift-header-avatar_omdbug.jpg 250w"
                />,
                "id": "header-avatar",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Circle image of middle aged man smiling on opaque palm tree gradient background.",
                  "sizes": "(max-width: 630px) 150px, 250px",
                  "src": "http://res.cloudinary.com/makeshift3ds/image/upload/makeshift-header-avatar_omdbug.jpg",
                  "srcSet": " http://res.cloudinary.com/makeshift3ds/image/upload/w_150/makeshift-header-avatar_omdbug.jpg 150w, http://res.cloudinary.com/makeshift3ds/image/upload/w_250/makeshift-header-avatar_omdbug.jpg 250w",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Full Stack Engineer / Designer",
                "className": "header-content",
              },
              "ref": null,
              "rendered": "Full Stack Engineer / Designer",
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
